### PR TITLE
[#LW-1936] Assistant back office : Master - Internal server error 500

### PR DESCRIPTION
**Pull request by parany**

### CODE REVIEW

File: modules/auth.js [hunk 1]
```diff
@@ -30,24 +31,61 @@ Queries the /validate endpoint and fetches two key values we need:
   This object doesn't exist until *after* the login has taken place,
   the first request will complain about a X-APPLE-WEBAUTH-TOKEN cookie
 */
-function refresh_validate(session, cb) {
+function refresh_validate(session) {
```
Comment (DelphinPlano) [inline modules/auth.js]: missing JSDoc

File: modules/auth.js [hunk 2]
```diff
@@ -30,24 +31,61 @@ Queries the /validate endpoint and fetches two key values we need:
...
+        resolve({ ok: true });
+      }
+    });
+  });
+}
 
-    cb(null);
+/**
+ * Log out current user with current dsid
+ * @param {Object} params
```
Comment (DelphinPlano) [inline modules/auth.js]: please, remove `params`

File: modules/auth.js [hunk 3]
```diff
@@ -30,24 +31,61 @@ Queries the /validate endpoint and fetches two key values we need:
...
   });
 }
 
 const auth = module.exports = {}
 
+/**
+ * Get basic session
+ * Make sure the previous user is logged out
+ */
+module.exports.getAsyncBasicSession = function() {
```
Comment (DelphinPlano) [inline modules/auth.js]: please, uise arrow function

File: modules/auth.js [hunk 4]
```diff
@@ -60,39 +98,39 @@ module.exports.getBasicSession = function() {
...
-                json: data
-            }, function(err, resp, data) {
-                if (err || data.error) 
-                    return reject("Invalid email/password combination.");
-                
-                // refresh after login
-                refresh_validate(session, (err, res) => {
-                    if(err) reject(err);
-                    else {
+        auth.getAsyncBasicSession(apple_id, password).then(session => {
```
Comment (DelphinPlano) [inline modules/auth.js]: wrong indentation

**Review by DelphinPlano — CHANGES_REQUESTED** [review]

**Comment by parany** [comment]
@DelphinPlano Please review it again.

**Review by DelphinPlano — APPROVED** [review]

**Comment by parany** [comment]
@vincentjacques Please review this PR.

**Pull request merged**